File: pysfmllib/Sprite.cs
using SFML.Graphics;
using SFML.System;

namespace pysfmllib
{
    public class Sprite
    {
        private Texture texture = null!;
        public RectangleShape rect;

        public Sprite(float width, float height)
        {
            rect = new(new Vector2f(width, height))
            {
                FillColor = Color.White
            };
        }
        public void LoadTexture(string path)
        {
            if (!File.Exists(path)) throw new FileNotFoundException($"Texture not found: {path}");

            texture = new Texture(path);
            rect.Texture = texture;
        }
        public void SetTextureRect(int x, int y, int w, int h)
        {
            rect.TextureRect = new IntRect(new Vector2i(x, y), new Vector2i(w, h));
        }
        public void SetSize(float w, float h) => rect.Size = new Vector2f(w, h);
        public Vector2f GetSize() => rect.Size;
        public void SetOrigin(float x, float y) => rect.Origin = new Vector2f(x, y);
        public Vector2f GetOrigin() => rect.Origin;
        public float SetRotation(float angle) => rect.Rotation = angle;
        public float GetRotation() => rect.Rotation;
        public void SetScale(float x, float y) => rect.Scale = new Vector2f(x, y);
        public void FlipX()
        {
            Vector2f size = rect.Size;
            Vector2f origin = rect.Origin;

            if (rect.Scale.X > 0)
            {
                rect.Scale = new Vector2f(-1, rect.Scale.Y);
                rect.Origin = new Vector2f(size.X, origin.Y);
            } else
            {
                rect.Scale = new Vector2f(1, rect.Scale.Y);
                rect.Origin = new Vector2f(0, origin.Y);
            }
        }
        public void FlipY()
        {
            Vector2f size = rect.Size;
            Vector2f origin = rect.Origin;

            if (rect.Scale.Y > 0)
            {
                rect.Scale = new Vector2f(rect.Scale.X, -1);
                rect.Origin = new Vector2f(origin.X, size.Y);
            }
            else
            {
                rect.Scale = new Vector2f(rect.Scale.X, -1);
                rect.Origin = new Vector2f(origin.X, 0);
            }
        }
        public Vector2f SetPosition(float x, float y)
        {
            rect.Position = new Vector2f(x, y);

            return rect.Position;
        }
        public Vector2f GetPosition() => rect.Position; 
        public void SetFillColor(byte r, byte g, byte b, byte a) => rect.FillColor = new Color(r, g, b, a);
        public void Draw(RenderWindow win) => win.Draw(rect);
    }
}

File: pysfml/sprite.py
import os
from pysfml._core import CsSprite, System

class Sprite:
  def __init__(self, width, height):
    try:
      f_width = System.Single(width)
      f_height = System.Single(height)
      self._x = 0.0
      self._y = 0.0
      self._w = float(width)
      self._h = float(height)
      self._obj = System.Activator.CreateInstance(CsSprite, f_width, f_height)
    except Exception as e:
      raise RuntimeError(f"Failed to create Sprite: {e}")

  def set_position(self, x, y):
    self.x, self.y = x, y
    return self._obj.SetPosition(System.Single(x), System.Single(y))
  
  def set_size(self, w, h):
    self._w = float(w)
    self._h = float(h)
    self._obj.SetSize(System.Single(w), System.Single(h))
  
  def set_fill_color(self, r,g,b,a=255):
    self._obj.SetFillColor(System.Byte(r), System.Byte(g), System.Byte(b), System.Byte(a))

  def set_texture_sprite(self, x, y, w, h):
    self._obj.SetTextureRect(x, y, w, h)

  def load_texture(self, path):
    abs_path = os.path.abspath(path)
    self._obj.LoadTexture(abs_path)

  def draw(self, win):
    win.draw(self)